### ISSUE TITLE
Bug: SIR 21512 characters

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#140]
User comment: Pls take a look at SIR 21512 unwound – characters are not recognized correctly

**Comment by danlyke** [comment]
Can we have a little more information? Is this in a cue sheet?

**Comment by mpogue2** [comment]
Email sent to user who made the comment...I will post the reply, or maybe he will post here!

**Comment by mpogue2** [comment]
Oh, I just noticed:  maybe this is the problem?
```
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=Generator content="Microsoft Word 12 (filtered)">
<title>Reach Out - I'll be there (The Four Tops) SIR 385</title>
<style>
```

"Reach Out" != "Unwound".

If that's the problem, this is a bug in the SqView cue sheet.

**Comment by mhkull** [comment]
I will return home tomorrow and provide screenshots and videos. Martin

**Comment by danlyke** [comment]
Or we need to be doing better (different?) character set handling.

**Comment by mhkull** [comment]
I upload 2 screenshots to the bleeding edge dropbox folder, fo you to look at. Pls delete screenshot, when you looked at them.

**Comment by mpogue2** [comment]
Thanks for the screenshot!  Posting it here, so we have a reference pic in the Issue itself.

![win_20170624_13_24_16_pro](https://user-images.githubusercontent.com/6316003/27519701-f0180c28-59ad-11e7-9722-135a3128aaa1.jpg)

**Comment by mpogue2** [comment]
Problem here is this line in the cue sheet file:
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">

This is probably the reason -- the file was converted to HTML by MS Word:
<meta name=Generator content="Microsoft Word 12 (filtered)">

So, the file is encoded in Windows-1252 format.  Those <?>'s mean that the layout engine doesn't understand the character.

What I found in there by inspecting in an iTerm window:
<A0> = NBSP (MS Word probably should have translated it to "&nbsp;")
<96> = long dash (MS Word probably should have translate it to "&ndash;")
<91> = SINGLE HIGH-REVERSED-9 QUOTATION MARK (MS Word should have translated it to "&#8219;", or maybe even just &#8216;)

When I display the HTML cue sheet in Safari, it renders properly.  My guess is that QTextEdit does NOT understand the WIN-1252 encoding, while Safari does.  Docs for QTextEdit say that it supports UTF-8 only (presumably even though the charset="" metadata tag is present).

I think that leaves us with 2 possibilities:
1) Do a character-by character substitution, as per the short list above, to convert 1252 to UTF-8 on-the-fly.  This is fragile, and will break again when we see more windows-1252 code points that were missed.  We could try to catch them all, with a longer list.
2) Run a script on the whole set of cue sheets to convert them all to UTF8.  This is probably the better solution.  a) We could do a one-time conversion, and redistribute all the files.  b) We could re-run the conversion, whenever new cue sheets were added, to catch them all.

None of these options is a great option.  I had this problem with database entries in the SqView DB, too.  Worse, they were Win1252 converted to UTF8 incorrectly, and it's impossible to undo the bad conversion on the fly.

**Comment by mpogue2** [comment]
Fixed by 96946f6dd170764f5ab38b3135bde31f59e57d11 .
I went with option 1, because I figured out a way to convert all such characters, using the built-in QTextCodec facility.  So, it's not fragile.  All such similar cue sheets should convert to Unicode properly now.

NOTE: If a file contains the phrase "windows-1252" anywhere, we will assume that the file is in 1252, and it will be converted to Unicode automatically.  I did not put in a further guard that looks to see if it was converted by MS WORD, specifically.  I also do not ensure that the presence of "windows-1252" is part of the metadata since no reasonable cue sheet would include that phrase in the body.

This is what it looks like after the proper conversion.  Yes, the layout is messed up, but that's a function of the MS Word converter that was used to create the HTML file (sorry!).

<img width="903" alt="win1252_fixed" src="https://user-images.githubusercontent.com/6316003/27520354-70cce23e-59be-11e7-855a-67b3fa6e1165.png">
